docs: Add comprehensive Device Discovery documentation (#692)
rokucommunity/vscode-brightscript-language: docs/device-discovery.md

---

## Common Scenarios

### Picking a Device After Launching VS Code

1. VS Code starts and loads any previously seen devices for the current network as `pending`.
2. You open the **Device Picker** (e.g., via `${promptForHost}` in your `launch.json`, or the **Select Device** command).
3. `refresh()` is called. If a scan is needed or the last scan was more than 30 minutes ago, an active SSDP scan begins.
4. While scanning, the picker shows existing (pending/online) devices immediately and a spinner indicates progress.
5. As devices respond, they appear in the list with an `online` state.
6. You select a device. The extension performs a quick health check. If the device is unreachable, an error is shown and the picker remains open.

### Discovering Devices After a Network Change

1. You connect to a new Wi-Fi network.
2. Within 3 minutes, the Network Change Monitor detects the change.
3. The device list is cleared and replaced with any cached devices for the new network (shown as `pending`).
4. `setScanNeeded()` is called.

> 5. The next time you open the Device Picker or the Devices Panel becomes visible, a fresh scan runs and populates the list.

Reviewer: ```suggestion
5. The next time you open the Device Picker or the Devices Panel becomes visible, a fresh scan runs and populates the list. If either view was already open, scan immediately
```

### What to Expect When Discovery Is Disabled

- No automatic scanning occurs.
- The Devices Panel is empty (no previously seen devices are loaded).
- The Device Picker shows no discovered devices.
- You must enter device IP addresses manually (e.g., via the **Enter IP address** option in the picker, or by setting `host` directly in `launch.json`).
- No network traffic from the extension related to device discovery.

### Resuming After Sleep

1. Your computer wakes from sleep.
2. The System Sleep Monitor detects the gap in timer execution and calls `setScanNeeded()`.

> 3. When you next open the Device Picker or the Devices Panel becomes visible, a fresh scan is triggered automatically.

Reviewer: ```suggestion
3. When you next open the Device Picker or the Devices Panel (or if either view was already open), a fresh scan is triggered automatically.
```
